From feaae2834f76edcfbe63e935edd66bb10ab34619 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Thu, 15 Sep 2016 22:40:48 +0100
Subject: [PATCH 10/12] wm: Fix negated logic in screen unredirect (screen
 tearing while recording)

Literally just that, we enabled for disabled, which is default, which has
in turn lead to people having very bad quality video recordings of Budgie.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 wm/wm.vala | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/wm/wm.vala b/wm/wm.vala
index 062a0dd..c075011 100644
--- a/wm/wm.vala
+++ b/wm/wm.vala
@@ -423,9 +423,9 @@ public class BudgieWM : Meta.Plugin
 
         var screen = this.get_screen();
         if (enab) {
-            Meta.Util.disable_unredirect_for_screen(screen);
-        } else {
             Meta.Util.enable_unredirect_for_screen(screen);
+        } else {
+            Meta.Util.disable_unredirect_for_screen(screen);
         }
         this.force_unredirect = enab;
     }
-- 
2.10.0

